### ISSUE TITLE
Unhandled Rejection Warning

### ISSUE THREAD

**Issue by zc73** [mrjackyliang/homebridge-adt-pulse#57]
### Describe the bug:
After more than a year of working perfectly, all of my sensors went unresponsive and I am repeatedly receiving the following message even after reinstalling the plugin: "WARNING unhandled rejection: TypeError: Cannot read property 'match' of null"
No other plugins in my environment have been affected.

### Steps to reproduce the bug:
1. 

### Expected behavior:
ADT Pulse sensors and panel functionality to return to normal in Homekit

### The Apple device you are using:
- Device: [e.g. iPhone 11 Pro]
- OS: [e.g. iOS 13.3]

### Homebridge information:
Find these 4 lines in the beginning of the logs, then replace the text inside the grave accents:

```sh
9/20/2021, 10:32:20 AM HOOBSADT Pulserunning on linux (arm)
9/20/2021, 10:32:20 AM HOOBSADT Pulsehomebridge-adt-pulse v2.0.4
9/20/2021, 10:32:20 AM HOOBSADT Pulsenode v14.17.6
9/20/2021, 10:32:20 AM HOOBSADT Pulsehomebridge v1.3.4
```

### Logs related to this bug:
Attempt to locate the bug in the logs, then replace `SAMPLE TEXT` inside the grave accents:

```sh
WARNING unhandled rejection: TypeError: Cannot read property 'match' of null
```
Current configuration:
"platform": "ADTPulse",
    "name": "ADT Pulse",
    "username": "xxxxxxxxxx@gmail.com",
    "password": "xxxxxxx",
    "country": "us",
    "removeObsoleteZones": true,
    "logActivity": true,
    "logLevel": "30"

**Comment by mrjackyliang** [comment]
Can you tell me which ADT version you are using? I know ADT has been fiercely making a lot of changes lately and breaking/slowing down the responses of the plugin.

**Comment by zc73** [comment]
I just logged into Pulse (first time in months) and there have been a lot
of UI changes. It looks like I’m on 21.0.0-354.

Thanks,
Z

**Comment by mrjackyliang** [comment]
Hmm, this is odd. I can't seem to replicate the error.

**Comment by mrjackyliang** [comment]
Can you turn on `debug` mode and give me the full dump of the error? Should include which file and line.

**Comment by zc73** [comment]
I turned on debug mode and bumped up Log Level to 50 and now I'm noticing
the login failed message... double checked my credentials and that's not
the issue.

9/22/2021, 7:41:43 PM Adt Pulse Bridge Adt Pulse Bridge starting
9/22/2021, 7:41:46 PM Adt Pulse Bridge [ DEBUG ] Initializing HAP-NodeJS
v0.9.4...
9/22/2021, 7:41:48 PM Adt Pulse Bridge Loaded plugin 'homebridge-adt-pulse'
9/22/2021, 7:41:48 PM Adt Pulse Bridge [ DEBUG ] Registering platform
'homebridge-adt-pulse.ADTPulse'
9/22/2021, 7:41:48 PM Adt Pulse Bridge Loading 1 platforms...
9/22/2021, 7:41:48 PM Adt Pulse Bridge ADT Pulse running on linux (arm)
9/22/2021, 7:41:48 PM Adt Pulse Bridge ADT Pulse homebridge-adt-pulse v2.0.4
9/22/2021, 7:41:48 PM Adt Pulse Bridge ADT Pulse node v14.17.6
9/22/2021, 7:41:48 PM Adt Pulse Bridge ADT Pulse homebridge v1.3.4
9/22/2021, 7:41:48 PM Adt Pulse Bridge ADT Pulse [ DEBUG ] Synchronizing
with ADT Pulse Web Portal...
9/22/2021, 7:41:48 PM Adt Pulse Bridge [ DEBUG ] Preparing Advertiser for
'Adt Pulse Bridge 50CC' using bonjour-hap backend!
9/22/2021, 7:41:48 PM Adt Pulse Bridge [ DEBUG ] Setup URI
'X-HM://0023ISYWY2HY6'
9/22/2021, 7:41:48 PM Adt Pulse Bridge [ DEBUG ] Starting to advertise 'Adt
Pulse Bridge 50CC' using bonjour-hap backend!
9/22/2021, 7:41:48 PM Adt Pulse Bridge Bridge is running on port 51836.
9/22/2021, 7:41:48 PM Adt Pulse Bridge [ DEBUG ] ADT Pulse: Logging in...
9/22/2021, 7:41:49 PM Adt Pulse Bridge [ DEBUG ] ADT Pulse: Response path
-> /myhome/21.0.0-354/access/signin.jsp
9/22/2021, 7:41:49 PM Adt Pulse Bridge [ DEBUG ] ADT Pulse: Response path
matches -> true
9/22/2021, 7:41:49 PM Adt Pulse Bridge [ DEBUG ] ADT Pulse: Web portal
version -> 21.0.0-354
9/22/2021, 7:41:50 PM Adt Pulse Bridge [ DEBUG ] ADT Pulse: Response path
-> /myhome/21.0.0-354/mfa/mfaSignIn.jsp?workflow=challenge
9/22/2021, 7:41:50 PM Adt Pulse Bridge [ DEBUG ] ADT Pulse: Response path
matches -> false
9/22/2021, 7:41:50 PM Adt Pulse Bridge [ ERROR ] ADT Pulse: Login failed.
9/22/2021, 7:41:50 PM Adt Pulse Bridge [ WARNING ] unhandled rejection:
TypeError: Cannot read property 'match' of null

**Comment by zc73** [comment]
Wait a sec... it looks like it's trying to hit multi-factor authentication,
which I don't have enabled in Pulse. Let me dig around a bit more in there
tonight.

**Comment by mrjackyliang** [comment]
Yeah, it seems like so. There isn't a way around 2FA for this plugin and lately it has been messing with a lot of my automations.

Worst yet, might be a good time to move to ADT Control.
